config_bucket.go

```diff
@@ -11,32 +11,45 @@ import (
 )
 
 const (
-	BucketTypeLocal = "local"
-	BucketTypeS3    = "s3"
+	BucketTypeGridFS = "gridfs"
+	BucketTypeLocal  = "local"
+	BucketTypeS3     = "s3"
 )
 
-// BucketConfig represents the admin config section for bucket storage.
+// BucketsConfig represents the admin config section for bucket storage.
+type BucketsConfig struct {
+	LogBucket BucketConfig `bson:"log_bucket" json:"log_bucket" yaml:"log_bucket"`
+}
+
+var bucketsConfigLogBucketKey = bsonutil.MustHaveTag(BucketsConfig{}, "LogBucket")
+
+// BucketConfig represents the admin config for an individual bucket.
 type BucketConfig struct {
-	LogBucket Bucket `bson:"log_bucket" json:"log_bucket" yaml:"log_bucket"`
+	Name   string `bson:"name" json:"name" yaml:"name"`
+	Type   string `bson:"type" json:"type" yaml:"type"`
+	DBName string `bson:"db_name" json:"db_name" yaml:"db_name"`
 }
 
-var bucketConfigLogBucketKey = bsonutil.MustHaveTag(BucketConfig{}, "LogBucket")
+func (c *BucketConfig) validate() error {
+	if c.Type == "" {
+		c.Type = BucketTypeS3
```

Comment:
Is this a reasonable default in all scenarios? I would have thought that S3 only applies in prod, we would explicitly want it to be S3, and in tests, we would explicitly want to set it to something other than S3. Instead of setting a default type, should we just validate that the type must be set to something before using the bucket (and have the UI default it to S3)?

Reply:
I did this as a safeguard for prod—I don't actually allow setting the bucket type in the UI to avoid someone accidentally changing it.